sdk/iot/Azure.Iot.Hub.Service/api/Azure.Iot.Hub.Service.netstandard2.0.cs

```diff
@@ -97,6 +99,23 @@ public partial class StatisticsClient
         public StatisticsClient() { }
     }
 }
+namespace Azure.Iot.Hub.Service.Authentication
+{
+    public partial interface ISasTokenProvider
+    {
+        string GetSasToken();
+    }
+    public partial class SasTokenProviderWithSharedAccessKey : Azure.Iot.Hub.Service.Authentication.ISasTokenProvider
+    {
+        public SasTokenProviderWithSharedAccessKey(string hostname, string sharedAccessPolicy, string sharedAccessKey, System.TimeSpan? timeToLive = default(System.TimeSpan?)) { }
+        public string GetSasToken() { throw null; }
+    }
+    public partial class StaticSasTokenProvider : Azure.Iot.Hub.Service.Authentication.ISasTokenProvider
```

Comment:
I am not sure if we should support connection string with sas token in it directly. The portal gives us a connection string with only the shared access key, and the [SDK board guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-auth-connection-strings) ask us to support the only those connection string formats that is readily available via portal.

The legacy client supports both formats.

Reply:
Let us bring this up to the team. I think we should only support what is readily available in the portal but let us conclude by talking to the team.

Reply:
You should also send an email to the SDK team before adding support for Connection string with sas token directly as it is not recommended. I'm not sure who the exact contact is. @drwill-ms who should we ask?

Reply:
When getting connection info from DPS, don't we get a hub url and a SAS token independently, and then use a helper to create a connection string? In that case, instead of creating a connection string, we could use this other method, right?

Reply:
@vinagesh Krzyszstof is our main contact for .NET questions

Reply:
The portal will will actually give you both the shared access key, and a connection string formatted with the shared access key and access policy. What it doesn't give you is the shared access signature (which is created by hashing the access key). So my approach was to take in the connection string directly.

Reply:
I'll reach out to Krzyszstof.

Reply:
I was thinking about device SDK again, so disregard my comment about DPS.